server/pulp/server/managers/content/query.py

```diff
@@ -290,7 +290,14 @@ def request_content_unit_file_path(self, content_type, relative_path):
         unit_path = os.path.join(self.get_root_content_dir(content_type), relative_path)
         unit_dir = os.path.dirname(unit_path)
         if not os.path.exists(unit_dir):
-            os.makedirs(unit_dir)
+            try:
```

Comment:
I propose that the conditional branch at 292 should go away altogether and that `os.makedirs` should be called in all cases. The try/except you've written will handle it correctly. The L292 check isn't working in all cases, and now we have two checks in a way for the same thing. Removing line 292 would reduce the complexity of this codepath. I just fixed a similar problem [in a recent PR](https://github.com/pulp/pulp/pull/1791/files#diff-6ceaf55d8c7a51c3a8243b1d577dcd11L103).